internal/security/sensitive_string.go

```diff
@@ -0,0 +1,55 @@
+package security
+
+import (
+	"encoding/json"
+)
+
+const RedactionPlaceholder = "***"
+
+// SensitiveString is a type for handling sensitive information securely.
+// This helps to achieve the following goals:
+//  1. Prevent accidental logging of sensitive information.
+//  2. Provide controlled visibility (e.g., redacted output for String() or MarshalJSON()).
+//  3. Enable controlled access to the sensitive value when needed.
+type SensitiveString struct {
+	value string
+}
+
+// NewSensitiveString creates a new SensitiveString
+func NewSensitiveString(value string) SensitiveString {
+	return SensitiveString{value: value}
+}
+
+// String provides a redacted version of the sensitive string
+func (s *SensitiveString) String() string {
```

Comment:
Because of immutable nature of `SensitiveString` I think value receiver makes more sense.
```suggestion
func (s SensitiveString) String() string {
```
^And for every other method as well.

Reply:
But.. then we'll copy it on each call 🤔 
I guess it's not a big deal though. There's just one `string` field, which value is not copied. So should be fine, will change 👍

Reply:
Well, `UnmarshalJSON` needs to be by pointer, doesn't work otherwise